src/ray/core_worker/core_worker.cc

```diff
@@ -1752,9 +1753,16 @@ Status CoreWorker::WaitPlacementGroupReady(const PlacementGroupID &placement_gro
   return status_future.get();
 }
 
-std::vector<rpc::ObjectReference> CoreWorker::SubmitActorTask(
+std::optional<std::vector<rpc::ObjectReference>> CoreWorker::SubmitActorTask(
     const ActorID &actor_id, const RayFunction &function,
     const std::vector<std::unique_ptr<TaskArg>> &args, const TaskOptions &task_options) {
+  /// Check whether backpressure may happen at the very beginning of submitting a task.
```

Comment:
talked with @MissiontoMars, this is actually not correct under multi-thread, as increasing counter and check is not atomic. so we might under counting or over counting the pending requests.

the question is: do we expect multiple thread submitted tasks to the same actor? cc @rkooo567 @ericl

Reply:
I don't think this is guaranteed to be called on a single thread. It is used by `submit_actor_task`, which can totally be used by multi threaded code.

Reply:
Dicussed with @scv119, we prefer to add a lock in `CoreWorker::SubmitActorTask` to guarantee thread safe.

Reply:
Hmm I am confused about this change. None of field is annotated with this lock that it's protected by this. Isn't it error prone in the future change?

Reply:
Also, to test this, we can have 10~20 python threads that submit to the actor task with backpressure limit 2 or sth like that.

Reply:
@rkooo567 the change is to ensure calling `PendingTaskFull` and `IncreaseCounter` are always atomic. yeah it's a little bit fragile but there is no good easy alternative. One big issue is actor_handle has an internal counter to ensure the sequential order of the request.